ckanext/dataset_subscriptions/actions/whatsapp_notifications.py

```diff
@@ -96,3 +110,77 @@ def _validate_plugin_extras(extras):
     for field in CUSTOM_FIELDS:
         out_dict[field['name']] = extras.get(field['name'], field['default'])
     return out_dict
+
+
+def get_phonenumber(user_dict):
+    if "activity_streams_whatsapp_notifications" in user_dict and "phonenumber" in user_dict:
+        if user_dict["activity_streams_whatsapp_notifications"] and user_dict["phonenumber"]:
+            phonenumber = user_dict["phonenumber"]
+            return phonenumber
+    return False
+
+
+def send_whatsapp_notifications(context, data_dict):
+    context = {'model': model, 'session': model.Session, 'ignore_auth': True}
+    users = logic.get_action('user_list')(context, {'all_fields': True})
+    for user in users:
+        user = logic.get_action('user_show')(context, {'id': user['id'],
+                                                       'include_plugin_extras': False})
+        if get_phonenumber(user):
+            return prepare_whatsapp_notifications(user)
+
+
+def prepare_whatsapp_notifications(user):
+    whatsapp_notifications_since = toolkit.config.get(
+            'ckanext.dataset_subscriptions.whatsapp_notifications_hours_since', 1)
+    whatsapp_notifications_since = int(whatsapp_notifications_since)
+    whatsapp_notifications_since = timedelta(hours=whatsapp_notifications_since)
+    whatsapp_notifications_since = (datetime.utcnow()
+                                    - whatsapp_notifications_since)
```

Comment:
mutating a local variable is a bad practise IMO. it's better to use different names for each of the steps. Also the logic is complicated enough that it should be a separate function. I suggest something like this:
```
def _whatsapp_notification_time_delta_utc():
    since_hours = int(toolkit.config.get(
            'ckanext.dataset_subscriptions.whatsapp_notifications_hours_since', 1))
    since_delta = timedelta(hours=whatsapp_notifications_since)
    since_datetime = (datetime.utcnow() - whatsapp_notifications_since_delta)
    return since_datetime
                                    
```